### PR TITLE
Add io.github.unrud.RecentFilter

### PR DESCRIPTION
**Pull request by Unrud**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### Permissions

* `--filesystem=xdg-data`: The application needs write access to `~/.local/share/recently-used.xbel` for its core function (Excluding files and folders from recently used files).
* `--filesystem=home:ro`: This is required for convenience. It allows the user to select folders via a file dialog, which are then added to the ignore pattern. The application needs the absolute paths on the host file system.

### CODE REVIEW

**Comment by Unrud** [comment]
bot, build

**Comment by Unrud** [comment]
bot, build io.github.unrud.RecentFilter

File: io.github.unrud.RecentFilter.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "44",
+    "sdk": "org.gnome.Sdk",
+    "command": "recent-filter",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--filesystem=xdg-data",
```
Comment (hfiguiere) [inline io.github.unrud.RecentFilter.json:12]: this permission is not allowed by default

Comment (Unrud) [reply io.github.unrud.RecentFilter.json:12]: Yes, I created a PR for an exception. The permission is required.

**Comment by Unrud** [comment]
bot, build io.github.unrud.RecentFilter

File: io.github.unrud.RecentFilter.json (hunk 1 above)
Comment (Unrud) [reply io.github.unrud.RecentFilter.json:12]: Exception got merged.

**Pull request closed without merging**